File: backend/src/main/java/com/inventory/backend/repository/InventoryAssetRepository.java
package com.inventory.backend.repository;

import com.inventory.backend.entity.InventoryAsset;
import com.inventory.backend.entity.InventoryAssetStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface InventoryAssetRepository extends JpaRepository<InventoryAsset, Long> {
    boolean existsByInventoryCodeIgnoreCase(String inventoryCode);

    boolean existsByInventoryCodeIgnoreCaseAndIdNot(String inventoryCode, Long id);

    Optional<InventoryAsset> findByInventoryCodeIgnoreCase(String inventoryCode);

    List<InventoryAsset> findAllByOrderByNameAsc();

    List<InventoryAsset> findByDepartmentIdOrderByNameAsc(Long departmentId);

    List<InventoryAsset> findByRoomIdOrderByNameAsc(Long roomId);

    List<InventoryAsset> findByStatusOrderByNameAsc(InventoryAssetStatus status);
}

File: backend/src/main/kotlin/com/inventory/backend/repository/InventoryAssetRepository.kt
package com.inventory.backend.repository

import com.inventory.backend.entity.InventoryAsset
import com.inventory.backend.entity.InventoryAssetStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface InventoryAssetRepository : JpaRepository<InventoryAsset, Long> {
    fun existsByInventoryCodeIgnoreCase(inventoryCode: String): Boolean
    fun existsByInventoryCodeIgnoreCaseAndIdNot(inventoryCode: String, id: Long): Boolean
    fun findByInventoryCodeIgnoreCase(inventoryCode: String): Optional<InventoryAsset>
    fun findAllByOrderByNameAsc(): List<InventoryAsset>
    fun findByDepartmentIdOrderByNameAsc(departmentId: Long): List<InventoryAsset>
    fun findByRoomIdOrderByNameAsc(roomId: Long): List<InventoryAsset>
    fun findByStatusOrderByNameAsc(status: InventoryAssetStatus): List<InventoryAsset>
}
